### PR TITLE
Upgrade machine image build system

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
This pull request was originally meant to publish our custom machine images on all the supported cloud providers. It has been repurposed to commit into this repository what we already have in production.

### Outdated checklist

* [x] ~Get rid of [`clone-amis.go`](https://github.com/iterative/terraform-provider-iterative/blob/8d6f803f1d02e152e7bb8dc39a006a3466ed91f8/cml/clone-amis.go) in favor of Packer's inbuilt multi-region support~
* [x] ~Migrate from JSON to HCL2, the new recommended format for Packer configuration files~
* [x] ~Tag all as many AWS resources as possible and use them for the Packer role policies~
* [x] ~Let Packer delete all the old images and leave only the latest one for each region~
* [x] ~Implement experimental support for Azure and Google Cloud~
* [x] ~Create a GitHub Actions workflow for testing and deploying images~
* [x] ~Publish Azure images; #197~
* [x] ~[Publish Google Cloud images](https://cloud.google.com/compute/docs/images/managing-access-custom-images#share-images-publicly), part of #196~

### CODE REVIEW

**Comment by DavidGOrtega** [comment]
> Get rid of clone-amis.go in favor of Packer's inbuilt multi-region support

Originally I tried with Packer but I was not able to clone, as far as I remember but build in every region thing that was impossible due to the credentials we had at that time

File: images/provisioner/setup.sh [hunk 1]
```diff
@@ -30,5 +30,5 @@ if [ ! -f "$FILE" ]; then
   sudo apt update && sudo apt install -y nvidia-docker2
   sudo systemctl restart docker
 
-  sudo echo 'OK' >"$FILE"
+  echo OK | sudo tee "$FILE"
```
Comment (DavidGOrtega) [inline images/provisioner/setup.sh:33]: was not this addressed at #205 ?

Comment (0x2b3bfa0) [reply images/provisioner/setup.sh:33]: Yes, please notice that this is a **draft** pull request and may include temporary upstream patches. 🙃

**Comment by 0x2b3bfa0** [comment]
> _Originally I tried with Packer but I was not able to clone, as far as I remember but build in every region thing that was impossible due to the credentials we had at that time_

Not sure about how it was originally, but this approach builds the image once and then clones the result, exactly like in our custom program.

**Pull request merged**